### PR TITLE
Fix printing multivariate polynomial ring with many variables

### PR DESCRIPTION
**Pull request by albinahlback**
The line above will only join `", "` in between each symbol, not after the last one.

### CODE REVIEW

**Comment by thofma** [comment]
I think this is fine, but @fingolfin should confirm

File: src/MPoly.jl [hunk 1]
```diff
@@ -570,7 +570,7 @@ function show(io::IO, ::MIME"text/plain", p::MPolyRing)
   print(io, " in ", ItemQuantity(nvars(p), "variable"), " ")
   if n > max_vars
     join(io, symbols(p)[1:max_vars - 1], ", ")
-    println(io, "..., ", symbols(p)[n])
+    println(io, ", ..., ", symbols(p)[n])
```
Comment (fingolfin) [inline src/MPoly.jl:573]: This seems right.

However, the exact same code appears in multiple places, so perhaps we should fix them all at once?
```
src/AbsMSeries.jl:125:    println(io, "..., ", symbols(p)[n])
src/FreeAssAlgebra.jl:73:    println(io, "..., ", symbols(a)[n])
src/LaurentMPoly.jl:42:    println(io, "..., ", symbols(p)[n])
src/MPoly.jl:573:    println(io, "..., ", symbols(p)[n])
```

Actually at least two of these have further issues:
```patch
diff --git a/src/AbsMSeries.jl b/src/AbsMSeries.jl
index 33782bf97..bf3811db2 100644
--- a/src/AbsMSeries.jl
+++ b/src/AbsMSeries.jl
@@ -119,7 +119,7 @@ function show(io::IO, ::MIME"text/plain", p::MSeriesRing)
   max_vars = 5 # largest number of variables to print
   n = nvars(p)
   print(io, "Multivariate power series ring")
-  print(io, "in ", ItemQuantity(nvars(p), "variable"), " ")
+  print(io, " in ", ItemQuantity(nvars(p), "variable"), " ")
   if n > max_vars
     join(io, symbols(p)[1:max_vars - 1], ", ")
     println(io, "..., ", symbols(p)[n])
diff --git a/src/LaurentMPoly.jl b/src/LaurentMPoly.jl
index 93516b526..c8a6b35e2 100644
--- a/src/LaurentMPoly.jl
+++ b/src/LaurentMPoly.jl
@@ -36,7 +36,7 @@ function show(io::IO, ::MIME"text/plain", p::LaurentMPolyRing)
   max_vars = 5 # largest number of variables to print
   n = nvars(p)
   print(io, "Multivariate Laurent polynomial ring")
-  print(io, "in ", ItemQuantity(nvars(p), "variable"), " ")
+  print(io, " in ", ItemQuantity(nvars(p), "variable"), " ")
   if n > max_vars
     join(io, symbols(p)[1:max_vars - 1], ", ")
     println(io, "..., ", symbols(p)[n])
```

I guess we should add some tests for those types that test the actual output, just like we do for maps in the "Generic.Map.printing" testset.

Comment (fingolfin) [reply src/MPoly.jl:573]: I've made some adjustments to this PR but did not yet attempt to add a dedicated test.

**Comment by albinahlback** [comment]
I think this should be okay. I suppose that more tests need be implemented, but I currently do not have the time to do so.

File: docs/src/mpolynomial.md [hunk 2]
```diff
@@ -96,6 +96,10 @@ julia> derivative(k, 1)
 julia> derivative(k, 2)
 0
 
+julia> R, x = polynomial_ring(ZZ, 10); display(R)
```
Comment (fingolfin) [inline docs/src/mpolynomial.md:99]:
```suggestion
julia> R, x = polynomial_ring(ZZ, 10); R
```

**Comment by fingolfin** [comment]
Thanks @albinahlback, I appreciate your contribution, and no need to do more! I just applied a minor tweak (can't use `display` in a doctest, as it always goes to the "regular" stdout and is not intercepted by the doctests).  And it is good to see that there actually was a "broken" test which is now "fixed" :-)

**Review by fingolfin — APPROVED** [review]

**Comment by thofma** [comment]
Hm, this was of course breaking

**Pull request merged**